### PR TITLE
[rosetta] subaccounts for pending and unpending active/inactive balances

### PR DESCRIPTION
**Pull request by supriyam9**
### Description
add subaccounts to track the following balances:
- `active_stake`
- `pending_active_stake`
- `inactive_stake`
- `pending_inactive_stake`

### CODE REVIEW

File: crates/aptos-rosetta/src/account.rs [hunk 1]
```diff
@@ -172,7 +179,171 @@ async fn get_balances(
                                 currency: native_coin(),
                             })
                         }
-                    } /* TODO: Right now operator stake is not supported
+                    } else if account.is_pending_active_stake() {
+                        // For total pending active stake, collect all underlying staking contracts and combine
+                        let mut total_pending_active: Option<u64> = None;
+                        maybe_operators = Some(vec![]);
```
Comment (xiaoxiaff) [inline crates/aptos-rosetta/src/account.rs:185]: What's the meaning of keeping the `maybe_operators`?

File: crates/aptos-rosetta/src/account.rs [hunk 2]
```diff
@@ -172,7 +179,171 @@ async fn get_balances(
...
+                        }
+
+                        if let Some(balance) = total_pending_inactive {
+                            balances.push(Amount {
+                                value: balance.to_string(),
+                                currency: native_coin(),
+                            })
+                        }
+                    } else if account.is_inactive_stake() {
+                        // For total inactive stake, collect all underlying staking contracts and combine
```
Comment (xiaoxiaff) [inline crates/aptos-rosetta/src/account.rs:306]: Small nits:
- Looks like multiple `else if` can be replace with the `match`. But not sure if that makes the code looks worse if multiple `match` exist in different scope and feel free to push back.
- These four types balances looks very similar in terms of looping contracts and `get_stake_balances` so we can merge them together to reduce duplicate code and only differentiate the desire field in  `get_stake_balances` result?

File: crates/aptos-rosetta/src/account.rs (hunk 1 above)
Comment (supriyam9) [reply crates/aptos-rosetta/src/account.rs:185]: i could be understanding this wrong, but it's looping through all staking contracts and totaling the balances by contract pool address. the `maybe_operators` is also one of the return values.

File: crates/aptos-rosetta/src/account.rs (hunk 2 above)
Comment (supriyam9) [reply crates/aptos-rosetta/src/account.rs:306]: yea i tried multiple match in different scope, but it looked very hard to read.
good idea, i'll differentiate desired field in `get_stake_balances`

File: testsuite/smoke-test/src/rosetta.rs [hunk 3]
```diff
@@ -477,6 +511,23 @@ async fn create_staking_contract(
...
+    account: &mut LocalAccount,
+    operator: AccountAddress,
+    amount: u64,
+) -> Response<Transaction> {
+    let unlock_stake = info
+        .transaction_factory()
+        .payload(aptos_stdlib::staking_contract_unlock_stake(
+            operator, amount,
+        ))
+        .sequence_number(2);
```
Comment (gregnazario) [inline testsuite/smoke-test/src/rosetta.rs:525]: This shouldn't be hardcoded, as it'll be easy to break.  Could you pass it into this function?

File: crates/aptos-rosetta/src/account.rs [hunk 4]
```diff
@@ -136,55 +136,53 @@ async fn get_balances(
...
-                    if account.is_total_stake() {
-                        // For total stake, collect all underlying staking contracts and combine
-                        let mut total_stake: Option<u64> = None;
-                        maybe_operators = Some(vec![]);
-                        for (operator, contract) in store.staking_contracts {
-                            // Keep track of operators
-                            maybe_operators.as_mut().unwrap().push(operator);
-                            match get_total_stake(
+                    let mut total_requested_balance: Option<u64> = None;
+                    maybe_operators = Some(vec![]);
```
Comment (gregnazario) [inline crates/aptos-rosetta/src/account.rs:140]: It also may be cleaner to do use the operators directly so you don't have to do the `as_mut().unwrap().push(operator)`

```
let mut operators = vec![];

...
operators.push(operator);

...

maybe_operators = Some(operators)
```

File: crates/aptos-rosetta/src/account.rs [hunk 5]
```diff
@@ -136,55 +136,53 @@ async fn get_balances(
...
+                            &account,
+                            contract.pool_address,
+                            version,
+                        )
+                        .await
+                        {
+                            Ok(Some(balance)) => {
+                                total_requested_balance = Some(
+                                    total_requested_balance.unwrap_or_default()
+                                        + u64::from_str(&balance.value).unwrap_or_default(),
```
Comment (gregnazario) [inline crates/aptos-rosetta/src/account.rs:155]: This seems complex for what it is, but I wrote it so I guess maybe there was a reason 🤷🏻‍♂️

**Review by gregnazario — APPROVED** [review]

**Review by 0xmaayan — APPROVED** [review]

**Pull request merged**